projects/vdk-core/src/vdk/internal/builtin_plugins/run/cli_run.py

```diff
@@ -80,6 +77,8 @@ def run_job(
                 + " logs for details and ensure the prerequisite for the failed component (details in stacktrace).",
                 exception=e,
             )
+        if execution_result.is_failed() and execution_result.get_exception():
```

Comment:
Shouldn't there be a check if execution_result is not None?

Reply:
It cannot be really None .
a) Either it is initialized in the `try:` block
b) or in `except` block `log_and_rethrow` would raise an exception

 I am setting `execution_result = None` because python linter/compiler does not realize  `log_and_rethrow` always throws and complains.

Reply:
Yes, I somehow skipped the part that log_and_rethrow always raises and exception.